tsdb/engine/tsm1/data_file.go

```diff
@@ -1194,12 +1139,14 @@ func (m *mmapAccessor) read(key string, timestamp time.Time) ([]Value, error) {
 		return nil, nil
 	}
 
-	return m.readBlock(entry)
+	return m.readBlock(entry, nil)
```

Comment:
The `read(key,time)` func doesn't pass one in currently and it'll get created down below if it's nil.  The `read` call is not used in the query path... just `readBlock` which does pass in a buffer now.